pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -61,13 +63,41 @@ makeWrapper() {
             command="${params[$((n + 1))]}"
             n=$((n + 1))
             echo "$command" >> "$wrapper"
-        elif [[ ("$p" == "--suffix") || ("$p" == "--prefix") ]]; then
+        elif [[ ("$p" == "--suffix") || ("$p" == "--prefix") || ("$p" == "--suffix-dedup") || ("$p" == "--prefix-dedup") ]]; then
             varName="${params[$((n + 1))]}"
             separator="${params[$((n + 2))]}"
             value="${params[$((n + 3))]}"
             n=$((n + 3))
             if test -n "$value"; then
-                if test "$p" = "--suffix"; then
+                if [[ "$p" == *'-dedup' ]]; then
+                    {
+                        # This abomination removes all occurences of the value that is to be prepended/appended.
+                        # We add (or re-add) the value later. This ensures it's always the last or first value.
+                        # As an example, adding /bin to a colon-separated environment variable would remove:
+                        # - /bin:*
+                        # - *:/bin:*
+                        # - *:/bin
+                        echo "OLDIFS=\$IFS"
```

Comment:
A style note: I recommend a here doc string, it will be a lot more readable.

Reply:
What's a doc string? This is bash, not Python

Reply:
I mean https://tldp.org/LDP/abs/html/here-docs.html

Reply:
Heredocs have weird formatting in bash and it is not very usual to use them for comments.

Reply:
Yeah, I thought about it, but here-docs don't handle indentation very well. I'll try it out and see which feels most readable in the end.

Reply:
I've added a commit that switches to using a heredoc. I'm pretty neutral to whether it's better or not, though. Less clutter, but worse indentation. However, if we don't care about how the resulting file looks at all, we could just indent the code in the heredoc.

Reply:
If the code in the wrapper becomes short enough, then I'd agree with removing the heredoc (I forgot about indentation!).